mscore/editstyle.cpp

```diff
@@ -665,6 +673,39 @@ EditStyle::EditStyle(Score* s, QWidget* parent)
       cs->startCmd();
       }
 
+//---------------------------------------------------------
+//   showEvent
+//---------------------------------------------------------
+
+void EditStyle::showEvent(QShowEvent* ev)
+      {
+      if (isTooWide || isTooHigh) { // the default size is bigger than the screen
+            // MuseScore might have restored a smaller size than the default,
+            // which is the only size that the constructor sees. So the size
+            // must be checked here too in order to allow MuseScore to restore
+            // the user's last size.
+            QRect scr = QGuiApplication::primaryScreen()->geometry();
+            QRect dlg = this->frameGeometry();
+            if (dlg.width() > scr.width() || dlg.height() > scr.height()) {
+                  QRect inner = this->geometry(); // excludes borders and title bar
+                  int w = scr.width() - (dlg.width() - inner.width()); 
+                  int h = scr.height() - (dlg.height() - inner.height());
+                  if (inner.width() > w)
+                        inner.setWidth(w);
+                  if (inner.height() > h)
+                        inner.setHeight(h);
+                  this->setMinimumSize(inner.width() / 2, inner.height() / 2);
```

Comment:
Also these calculations don't look necessary to setup a reasonable minimal dialog width. We don't need to define any precise limit here, some approximate assessment would be more than enough.

Reply:
The idea is to restore the original size.  If you resize the window while the pageList is hidden, for example, you don't want to resize the whole dialog to the default size all over again.

Reply:
Look at the other PR, #4938.  That illustrates the rest of this stuff.

Reply:
This dialog is excessively complex in terms of sizing.  Everything interacts and resizes automatically.  This code manages to preserve the existing user size of the dialog each time you show/hide the page list, and also not to modify the preset minimum size for the dialog, which should be preserved always.

Reply:
I misunderstood your comment the first time I read it.  I deleted those responses..So you want to put literals here?  I copied this divide by 2 logic from your patch's link at the top of this PR's page.  Tell me what you want to do here and I'll do it.  Personally, I would set the limits at 50 width and 80 height, to let the user size it to whatever size they like.  The height should be larger than the width because of the title bar.

Reply:
In practice, no user on a small screen is going to want to size the dialog smaller once it is sized properly for their screen.  As you say, so long as the value is reasonable, there is no issue.  The dialog is only sizable smaller when the user's screen is too small, so it's a very limited user case.

Reply:
I think you're missing the point of these calculations.  They are not for setting the minimum size.  They are for setting the initial size so that it fits in the screen.  The key line of code is not `setMinimumSize()` it is `setGeometry()` below there.

Reply:
...but you are correct that I'm setting minimum size twice, if that's what you meant.  I do it here and in the constructor.  It must be in the constructor for restoreGeometry to succeed.  I can remove this one line of code here safely.